lib/httpResponse.ts

```diff
@@ -23,13 +23,18 @@ export interface HttpResponse {
    */
   headers: HttpHeaders;
 
+  /**
+   * The body of this HttpResponse after it has been deserialized.
+   */
+  deserializedBody?: any;
+
   /**
    * Get the body of this HttpResponse as a string.
    */
   textBody(): Promise<string | undefined>;
 
   /**
-   * Get the body of this HttpResponse after it has been deserialized.
+   * Get the body of this HttpResponse before it has been deserialized.
    */
-  deserializedBody(): Promise<any>;
+  serializedBody(): Promise<any | undefined>;
```

Comment:
I'd be happier with `rawBody` or `body` alongside `deserializedBody` so that we don't imply that the body content is always something deserializable. A response body stream also needs to be exposed, but I think that will happen later.

Reply:
How about calling it `jsonBody()`? It's the equivalent of calling `JSON.parse(await this.textBody())`, so that seems to fit to me.

Reply:
Eh... maybe `parsedBody()` would work. Funny, that's what you and Amar suggested at the beginning of this work.

Reply:
If you want to expose basically the result of JSON.parse, which I think the current runtime design does on its HttpOperationResponse type, parsedBody (since it might be parsed XML) is fine.

However since serialization is moving into the pipeline it seems like you might as well expose the response body in just 2 forms:
1. the raw form, probably a [ReadableStream](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream)
2. the parsed-then-deserialized (by an OperationSpec/TypeSpec) object tree version of the body

Reply:
Okay, okay. You're right. There really should only be two ways to get the body of the response:
1. `textBody()`
2. `deserializedBody()`
I've made this change.